src/selfie_park/src/park_server.cpp

```diff
@@ -198,142 +162,119 @@ void ParkService::drive(float speed, float steering_angle)
 
 bool ParkService::toParkingSpot()
 {
-  if (actual_parking_position_.x_ < back_wall_ + minimal_start_parking_x_)
-  {
-    drive(parking_speed_, 0.0);
-  }
-  else return true;
-
-  return false;
-}
-
-bool ParkService::park()
-{
-  switch (move_state_)
-  {
-    case first_phase:
-      if (state_msgs_) ROS_INFO_THROTTLE(5, "  1st phase");
-      drive(parking_speed_, -max_turn_);
-      if (actual_parking_position_.rot_ < -max_rot_)
-      {
-        move_state_ = straight;
-      }
-      if (actual_parking_position_.y_ < mid_y_)
-      {
-        move_state_ = second_phase;
-      }
-      break;
-
-    case straight:
-      if (state_msgs_) ROS_INFO_THROTTLE(5, "  straight");
-      drive(parking_speed_, 0.0);
-      if (actual_parking_position_.y_ < dist_turn_ + middle_of_parking_spot_y_)
-      {
-        move_state_ = second_phase;
-      }
-      break;
-
-    case second_phase:
-      if (state_msgs_) ROS_INFO_THROTTLE(5, "  2nd phase");
-      drive(parking_speed_, max_turn_);
-      if (actual_parking_position_.rot_ > 0.0 ||
-          (actual_front_parking_position_.x_ > front_wall_ - max_distance_to_wall_))
-      {
-        move_state_ = first_phase;
-        return true;
-      }
-      break;
-  }
-  return false;
-}
-
-bool ParkService::leave()
-{
-  switch (move_state_)
-  {
-    case first_phase:
-      if (state_msgs_) ROS_INFO_THROTTLE(5, "  1st phase");
-      drive(parking_speed_, max_turn_);
-      if (actual_parking_position_.rot_ > max_rot_)
-      {
-        move_state_ = straight;
-      }
-      if (actual_parking_position_.y_ > mid_y_)
-      {
-        move_state_ = second_phase;
-      }
-      break;
-
-    case straight:
-      if (state_msgs_) ROS_INFO_THROTTLE(5, "  straight");
-      drive(parking_speed_, 0.0);
-      if (actual_parking_position_.y_ > leaving_target_ - dist_turn_)
-      {
-        move_state_ = second_phase;
-      }
-      break;
-
-    case second_phase:
-      if (state_msgs_) ROS_INFO_THROTTLE(5, "  2nd phase");
-      drive(parking_speed_, -max_turn_);
-      if (actual_parking_position_.rot_ < 0.0)
-      {
-        move_state_ = first_phase;
+    std::cout<<"target "<<back_target_<<" actual "<<actual_dist_<<std::endl;
+    if(actual_dist_ > back_target_)
+    {
         return true;
-      }
-      break;
-  }
-  return false;
+    }
+    drive(parking_speed_, 0.);
+    return false;
 }
 
-float ParkService::Position::quatToRot(const geometry_msgs::Quaternion &quat)
+bool ParkService::park()
 {
-  return static_cast<float>(tf::getYaw(quat));
-}
+    park_spot_dist_ -= 1./2.*std::sin(max_turn_)*std::abs(actual_dist_ - prev_dist_);
+    std::cout<<"sin "<<sin(max_turn_)<<std::endl;
+    std::cout<<"diff "<<actual_dist_ - prev_dist_<<std::endl;
+    std::cout<<"park spot dist "<<park_spot_dist_<<std::endl;
+    if(move_state_ == first_phase)
+    {
+        if(park_spot_dist_ > 0.)
+        {
+            if(actual_dist_ > front_target_)
```

Comment:
Tutaj jak już nie będzie front target to dałbym zmienną i dodawał do niej różnicę dystansu pomiędzy wywołaniami. Dla iteracji jazdy w przód sprawdzał czy ta zmienna osiągnęła już iter_distance, dla iteracji jazdy w tył sprawdzał czy nie jest mniejsza od 0. Dodatkowo w if'ie dodać zmienną bool określającą, która iteracja (przód czy tył) jest aktualnie wykonywana. 

Po wykonaniu parkowanie zmienić zmienną bool na przeciwną.

To sprawi, że jak skończymy powiedzmy na przejechaniu 0.05 w przód to przy wyjeżdżaniu zacznie od doprowadzenia tej zmiennej do 0 jazdą w tył. (czyli idealne odtworzenie sekwencji w tył).